src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -1618,23 +1618,24 @@ public static IEnumerable<object[]> Join_ObjectArray_TestData()
             yield return new object[] { null, new object[] { "Foo", "Bar", "Baz" }, "FooBarBaz" };
             yield return new object[] { "$$", new object[] { "Foo", null, "Baz" }, "Foo$$$$Baz" };
 
-            // Join does nothing if array[0] is null
-            yield return new object[] { "$$", new object[] { null, "Bar", "Baz" }, "" };
+            // Test join when first value is null
+            yield return new object[] { "$$", new object[] { null, "Bar", "Baz" }, "$$Bar$$Baz" };
```

Comment:
This will end up failing when run on desktop, right?  We'll need to separate it out into a separate test that's disabled for NetFramework.